code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/AEPPushTemplate.kt

```diff
@@ -41,276 +40,184 @@ internal sealed class AEPPushTemplate {
-                null
-            )
-        )
-        notificationImportance =
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) getNotificationImportanceFromString(
-                DataReader.optString(
-                    data,
-                    PushTemplateConstants.PushPayloadKeys.NOTIFICATION_PRIORITY,
-                    null
-                )
-            ) else null
-
-        notificationVisibility = NotificationVisibility.getNotificationCompatVisibilityFromString(
-            DataReader.optString(
-                data,
-                PushTemplateConstants.PushPayloadKeys.NOTIFICATION_VISIBILITY,
-                null
-            )
-        )
-        channelId =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.CHANNEL_ID, null)
-        templateType =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TEMPLATE_TYPE, null)
-                ?.let { PushTemplateType.fromString(it) }
-        tag = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TAG, null)
-        val stickyValue =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.STICKY, null)
-        isNotificationSticky =
-            if (stickyValue.isNullOrEmpty()) false else java.lang.Boolean.parseBoolean(
-                stickyValue
-            )
-        ticker = DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.TICKER, null)
-        expandedBodyText = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT, null
-        )
-        expandedBodyTextColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.EXPANDED_BODY_TEXT_COLOR, null
-        )
-        titleTextColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.TITLE_TEXT_COLOR, null
-        )
-        smallIconColor = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.SMALL_ICON_COLOR, null
-        )
-        notificationBackgroundColor = DataReader.optString(
-            data,
-            PushTemplateConstants.PushPayloadKeys.NOTIFICATION_BACKGROUND_COLOR,
-            null
-        )
-        isFromIntent = false
+    init {
+        // extract the notification payload version
+        payloadVersion = data.getString(PushPayloadKeys.VERSION)?.toInt()
+            ?: throw IllegalArgumentException("Required field \"${PushPayloadKeys.VERSION}\" not found.")
+
+        // extract the text information
+        title = data.getString(PushPayloadKeys.TITLE)
+            ?: throw IllegalArgumentException("Required field \"${PushPayloadKeys.TITLE}\" not found.")
+        body = data.getString(PushPayloadKeys.BODY)
+            ?: throw IllegalArgumentException("Required field \"${PushPayloadKeys.BODY}\" not found.")
+        expandedBodyText = data.getString(PushPayloadKeys.EXPANDED_BODY_TEXT)
```

Comment:
I am attempting to see if was can make this pattern more generic so that the text need not be repeated always.
How about using optString (for nullable values) and getString (to throw and log the error message when null)?

Reply:
Discused with Pravin on chat. We aligned on making the  methods `getString` and `getRequiredString`

Reply:
I got this updated